src/utils/featheryClient/integrationClient.ts

```diff
@@ -149,6 +156,104 @@ export default class IntegrationClient {
     );
   }
 
+  async triggerFlinksLoginId(loginId: string, updateFieldValues: any) {
+    await initFormsPromise;
+    const { userId } = initInfo();
+    const params = encodeGetParams({
+      form_key: this.formKey,
+      fuser_key: userId,
+      login_id: loginId
+    });
+
+    const url = `${API_URL}flinks/login-id/?${params}`;
+
+    let response: any = await this._fetch(url).catch((e) => {
+      console.error(
+        {
+          id: 'triggerFlinksLoginId',
+          message: 'failed to fetch flinks field values'
+        },
+        e
+      );
+      return null;
+    });
+
+    let pollInterval: NodeJS.Timeout;
+    let intervalCleared = false;
+
+    response = await new Promise((resolve, reject) => {
+      let innerResponse: any;
+
+      setTimeout(async () => {
+        if (!intervalCleared) {
+          clearInterval(pollInterval);
+          intervalCleared = true;
+          resolve(innerResponse);
+        }
+      }, this.FLINKS_TIMEOUT_MS);
+
+      pollInterval = setInterval(async () => {
+        innerResponse = await this._fetch(url).catch((e) => {
+          console.error(
+            {
+              id: 'triggerFlinksLoginId',
+              message: 'failed to fetch flinks field values'
+            },
+            e
+          );
+          return null;
+        });
+
+        if (innerResponse && (innerResponse as any).status === 200) {
+          clearInterval(pollInterval);
+          intervalCleared = true;
+          resolve(innerResponse);
+        } else {
+          reject(innerResponse);
+        }
+      }, this.FLINKS_REQUEST_RETRY_TIME_MS);
+    }).catch((err) => {
+      console.error(
+        {
+          id: 'triggerFlinksLoginId',
+          message: 'failed to fetch flinks field values'
+        },
+        err
+      );
+      return null;
+    });
+
+    if (!response) {
+      console.log({ id: 'triggerFlinksLoginId', message: 'response is null' });
+      return;
+    }
+
+    const responseData = await response?.json();
+
+    if (!responseData) {
+      console.log({
+        id: 'triggerFlinksLoginId',
+        message: 'responseData is null'
+      });
+      return;
+    }
+
+    if (responseData.field_values) {
+      const fieldValues = responseData.field_values;
+      console.log({
+        id: 'triggerFlinksLoginId',
+        message: 'updating field values',
+        fieldValues
+      });
+      updateFieldValues(fieldValues);
```

Comment:
should be done in flinks.ts rather than the client handler

Reply:
@bo-dun-1 Done. Will push the change shortly.